README_FRAME_FORMAT.rst

```diff
@@ -164,23 +146,33 @@ __________________________________________
 :tdecomp:
     (``int16``) Number of threads for decompression.  If 0, same than `dctx`.
 
+:map of metalayers:
+    This is a *msgpack-formattted* map for the different metalayers.  The keys will be a string (0xa0 + namelen) for
+    the names of the metalayers, followed by an int32 (0xd2) for the *offset* of the value of this metalayer.  The
+    actual value will be encoded as a bin32 (0xc6) value later in frame.
+
+Chunks
+------
 
-The chunks section
-------------------
+The chunks section is composed of one or more Blosc data chunks followed by an index chunk::
 
-Here there is the actual data chunks stored sequentially::
+    +========+========+========+========+===========+
+    | chunk0 | chunk1 |   ...  | chunkN | chunk idx |
+    +========+========+========+========+===========+
 
-    +========+========+========+===========+
-    | chunk0 | chunk1 |   ...  | chunk idx |
-    +========+========+========+===========+
+Each chunk is stored sequentially and follows the format described in the
+`chunk format <README_CHUNK_FORMAT.rst>`_ document.
 
-The different chunks are described in the `chunk format <README_CHUNK_FORMAT.rst>`_ document.  The `chunk idx` is an index for the different chunks in this section.  It is made by the 64-bit offsets to the different chunks and compressed into a new chunk, following the regular Blosc chunk format.
+The `chunk idx` is a Blosc chunk containing the indexes to each chunk in this section.  The data in the
+chunk is a list of 64-bit offsets to each chunk. The index chunk follows the regular Blosc chunk format and
+can be compressed.
 
 
-The trailer section
--------------------
+Trailer
+-------
 
-Here it is data that can change in size, mainly the `usermeta` chunk::
+The trailer for the frame is encoded via `msgpack <https://msgpack.org>`_ and contains a user meta data chunk and
+a fingerprint::
```

Comment:
I would add a note here stating that the `usermeta` chunk can change in size during the life of the frame.  This is an important feature, and the reason why `usermeta` is in the trailer, and not in the header.